### PR TITLE
#160577565 Setup react router version 4 (React-router-dom)

### PR DESCRIPTION
**Pull request by gitaumoses4**
**What does this PR do?**
Setups react router dom version 4 for the project.

**Description of Task to be completed?**

Create login, registration and index page to display the routing functionality.
Created AuthenticationRoute for protecting of the routes that are for authenticated users only.

**How should this be manually tested?**

Clone the repository to your local environment.
```
git clone https://github.com/andela/ah-codeblooded-frontend.git
```

Change directories
```
cd ah-codeblooded-frontend/
```

Install the dependencies
```
npm install
```

Run the server
```
yarn start
```

It will open a browser with the current route at `/`
Click on the `Login` and `Register` links to view the routing process

**Any background context you want to provide?**

None

**What are the relevant pivotal tracker stories?**

[#160577565](https://www.pivotaltracker.com/story/show/160577565)

**Screenshots (if appropriate)**
screen shot 2018-10-30 at 16 59 55
screen shot 2018-10-30 at 17 00 00
screen shot 2018-10-30 at 17 00 02

### CODE REVIEW

File: src/pages/SignUpPage/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+              </div>
+            </div>
+          </div>
+        </div>
+      </>
+    );
+  }
+}
+
+export default SignUpPage;
```
[the remark this answers is not in the document]
Comment (paulrimiru) [reply src/pages/SignUpPage/index.js:35]: Please look into this

[the remark this answers is not in the document]
Comment (gitaumoses4) [reply src/pages/SignUpPage/index.js:35]: Cool. Let me have a look at it.

File: src/pages/LoginPage/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { ROUTES } from '../../utils/routes';
+
+class LoginPage extends Component {
+  render() {
+    return (
+      <>
+        <nav className="white black-text">
+          <div className="container">
+            <div className="nav-wrapper">
+              <a href={ROUTES.index} className="brand-logo center black-text logo">Author's Haven</a>
```
Comment (paulrimiru) [inline src/pages/LoginPage/index.js]: Try to use this 
```
<a href={ROUTES.index} className="brand-logo center black-text logo">{`Author's Haven`}</a>
```
instead

File: src/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import {
+  BrowserRouter as Router, Redirect, Route, Switch,
```
Comment (paulrimiru) [inline src/routes/index.js]: Separate these into separate lines in order to improve readability.

File: src/routes/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  );
+
+  render() {
+    const { component: Comp, ...rest } = this.props;
+    return (
+      <Route
+        {...rest}
+        render={props => (
+          this.isAuthenticated
+            ? (this.renderIfAuthenticated(<Comp {...props} user={this.user} />))
```
Comment (paulrimiru) [inline src/routes/index.js]: Brackets here are not necessary, please remove them.

File: src/routes/index.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    ) : (
+      component
+    )
+  );
+
+  render() {
+    const { component: Comp, ...rest } = this.props;
+    return (
+      <Route
+        {...rest}
```
Comment (paulrimiru) [inline src/routes/index.js]: Add spaces between the brackets and the `rest` object.

File: src/routes/index.js (hunk 4 above)
Comment (paulrimiru) [reply src/routes/index.js]: Also pad props with spaces

File: src/pages/HomePage/index.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { ROUTES } from '../../utils/routes';
+
+class HomePage extends Component {
```
Comment (paulrimiru) [inline src/pages/HomePage/index.js:5]: Add prop validation

**Review by paulrimiru — CHANGES_REQUESTED** [review]

File: src/routes/index.js (hunk 5 above)
Comment (gitaumoses4) [reply src/routes/index.js]: For this one, the eslint automatically formats back to no spaces.

File: .eslintrc.json [hunk 7]
```diff
@@ -5,8 +5,10 @@
     "jest":true,
     "node":true
   },
-  "esversion":6,
-  "react/jsx-filename-extension": [1, { "extensions": [".js",".jsx"] }],
+  "rules":{
+    "react/jsx-filename-extension": [1, { "extensions": [".js",".jsx"] }],
+    "esversion":6
+  },
```
Comment (charisschomba) [inline .eslintrc.json:23]: Why did you make this change on this file?

Comment (gitaumoses4) [reply .eslintrc.json:23]: Oh yes, Actually the `esversion` and `react/jsz-filename-extension` cannot be in the root of the eslintrc file. It gives an error as seen from the hound review above.
```
- Unexpected top-level property "esversion".
```
They should be part of the `rules` in the eslintrc file.

File: src/routes/index.js (hunk 5 above)
Comment (paulrimiru) [reply src/routes/index.js]: Okay I see, can you give it a more descriptive name,  I think rest is not descriptive enough

Comment (gitaumoses4) [reply src/routes/index.js]: All done. 👍

**Review by paulrimiru — APPROVED** [review]

**Review by charisschomba — APPROVED** [review]

**Review by bevkololi — APPROVED** [review]
Good job here @gitaumoses4

**Pull request merged**